src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +18,44 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(phone);
+            transaction.commit();
+            return phone;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't add phone " + phone + " to Db ", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder cb = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = cb.createQuery(Phone.class);
+            Root<Phone> phoneRoot = query.from(Phone.class);
+            Predicate combinedPredicate = cb.and();
+            for (Map.Entry<String, String[]> stringEntry : params.entrySet()) {
+                CriteriaBuilder.In<String> partOfPredicate = cb.in(phoneRoot.get(stringEntry
+                        .getKey()));
+                for (String parameter : stringEntry.getValue()) {
+                    partOfPredicate.value(parameter);
+                }
+                combinedPredicate = cb.and(combinedPredicate, partOfPredicate);
+            }
+            return session.createQuery(query.where(combinedPredicate)).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't find all phones ", e);
```

Comment:
maybe you need to tell more about the exception

Reply:
I think, that's enough